business-networks-workflows/src/main/kotlin/net/corda/bn/flows/ApproveMembershipAttributeChangeFlow.kt

```diff
@@ -0,0 +1,96 @@
+package net.corda.bn.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.bn.contracts.ChangeRequestContract
+import net.corda.bn.states.ChangeRequestState
+import net.corda.bn.states.ChangeRequestStatus
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.flows.FlowException
+import net.corda.core.flows.InitiatingFlow
+import net.corda.core.flows.StartableByRPC
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.FlowSession
+import net.corda.core.identity.Party
+import net.corda.core.internal.uncheckedCast
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+
+/**
+ * This flow is initiated by an authorised member who wants to approve a [MembershipModificationRequest] to change
+ * a members attributes.
+ * Modifies an existing [ChangeRequestState]'s status to APPROVED on initiator's and authorised member's ledgers and
+ * executes the changes.
+ *
+ * @property requestId The ID of the request which needs to be accepted.
+ * @property notary Identity of the notary to be used for transactions notarisation. If not specified, first one from the whitelist will be used.
+ * */
+@InitiatingFlow
+@StartableByRPC
+class ApproveMembershipAttributeChangeFlow(
+        private val requestId: UniqueIdentifier,
+        private val notary: Party? = null
+) : MembershipManagementFlow<SignedTransaction>() {
+
+    @Suspendable
+    override fun call(): SignedTransaction {
+        val bnService = serviceHub.cordaService(BNService::class.java)
+
+        val membershipChangeRequest = bnService.getMembershipChangeRequest(requestId)
+                ?: throw MembershipChangeRequestNotFoundException("Could not find change request state with $requestId request ID")
+
+        val membershipId = membershipChangeRequest.state.data.membershipId
+
+        auditLogger.info("$ourIdentity started accepting and activating membership attribute changes of " +
+                "member with $membershipId membership ID for request with $requestId request ID")
+
+        bnService.getMembership(membershipId)
+                ?: throw MembershipNotFoundException("Membership state with $membershipId linear ID doesn't exist")
+
+        val membershipChangeData = membershipChangeRequest.state.data
+
+        // check whether party is authorised to initiate flow
+        val networkId = bnService.getMembership(membershipChangeData.membershipId)!!.state.data.networkId
+
+        if (membershipChangeData.pendingBusinessIdentityChange != null) {
+            authorise(networkId, bnService) { it.canModifyBusinessIdentity() }
+            // Call the modify business identity flow if the business identity has to be modified
+            subFlow(ModifyBusinessIdentityFlow(membershipId, membershipChangeData.pendingBusinessIdentityChange!!, notary))
```

Comment:
I'm not convinced committing the approved changes (running 1 or 2 sublfows) is required. This flow approves a request. We tried to write all flows to be building blocks or primitives for tooling or more complex cordapps. This flow does more than it says.